### PR TITLE
Remove option for using flash attention

### LINKED ISSUE #39: SDPA dispatch to TPU Pallas kernel is missing inverse sqrt scaling and overrides casual as always true
Hi @qihqi ,

I have encountered the following during experimentation. When using the pallas tpu kernel for scaled_dot_product_attention the following code is executed:
```
        return flash_attention.flash_attention(
            query, key, value, causal=True, block_sizes=block_sizes
        )
```

This always overrides causal as `True`, which is problematic for diffusion text-to-image models and does not match the [reference](https://github.com/google/torchax/blob/main/torchax/ops/jtorch.py#L128) implementation that handles this parameter correctly.

Another issue is the inverse square root scaling present in the reference [here](https://github.com/google/torchax/blob/main/torchax/ops/jtorch.py#L139) is not part of the pallas kernel computation, so it needs to be handled before the pallas call as is done for torch xla attention backend in diffusers [here](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/attention_dispatch.py#L1855).

The following replaces torch sdpa call with a correct tpu pallas flash attention call:
```
def use_tpu_flash_attention(env):
    env.config.use_tpu_flash_attention = True

    @jax.default_matmul_precision("default")
    def custom_attention(query, key, value, attn_mask=None, dropout_p=0.0,
                         is_causal=False, scale=None, enable_gqa=False):
        block_sizes = flash_attention.BlockSizes(
            block_b=min(2, query.shape[0]),
            block_q=min(512, query.shape[2]),
            block_k_major=min(512, key.shape[2]),
            block_k=min(512, key.shape[2]),
            block_q_major_dkv=min(512, query.shape[2]),
            block_k_major_dkv=min(512, key.shape[2]),
            block_k_dkv=min(512, key.shape[2]),
            block_q_dkv=min(512, query.shape[2]),
            block_k_major_dq=min(512, key.shape[2]),
            block_k_dq=min(256, key.shape[2]),
            block_q_dq=min(1024, query.shape[2]),
        )
        query = query / math.sqrt(query.shape[-1])
        jk, jq, jv = torchax.interop.jax_view((query, key, value))
        res = flash_attention.flash_attention(jk, jq, jv, block_sizes=block_sizes, causal=is_causal)
        return torchax.interop.torch_view(res)

    env.override_op_definition(torch.nn.functional.scaled_dot_product_attention, custom_attention)
```

This still misses correct handling of the attention mask, but fixes the two issues above.

I can open a PR with a fix for these two issues or wait for the core dev team to fix it. Thanks.

### PR DESCRIPTION
**Pull request by qihqi**
Will encourage the use of env.override_op_defition

fixes #39

### CODE REVIEW

**Comment by jialei777** [comment]
@qihqi Do you have example on using env.override_op_defition for gmm kernel with moe models?

**Comment by qihqi** [comment]
> @qihqi Do you have example on using env.override_op_defition for gmm kernel with moe models?

`override_op_definition` only works for overriding functions / operators defined in torch. I don't think there is one for gmm. So here the recommendation is to either call the pallas kernel directly in Python code, or make the gmm logic inside of a `torch.nn.Module` then at runtime iterate the module hierarchy and do live module-swap.
